tests/test_trailedSourceFilter.py

```diff
@@ -0,0 +1,85 @@
+import unittest
+from lsst.ap.association import TrailedSourceFilterTask
+import numpy as np
+import pandas as pd
+import lsst.afw.image as afwImage
+import lsst.utils.tests
+
+
+class TestTrailedSourceFilterTask(unittest.TestCase):
+
+    def setUp(self):
+        """Create sets of diaSources and diaObjects.
+        """
+        rng = np.random.default_rng(1234)
+        scatter = 0.1 / 3600
+        self.nSources = 5
+        self.diaSources = pd.DataFrame(data=[
+            {"ra": 0.04*idx + scatter*rng.uniform(-1, 1),
+             "dec": 0.04*idx + scatter*rng.uniform(-1, 1),
+             "diaSourceId": idx, "diaObjectId": 0, "trailLength": 5.5*idx}
+            for idx in range(self.nSources)])
+        visitInfo = afwImage.VisitInfo(exposureTime=30.0,
+                                       )
+        exposureInfo = afwImage.ExposureInfo()
+        exposureInfo.setVisitInfo(visitInfo)
+        maskedImage = afwImage.MaskedImageF(lsst.geom.Extent2I(64, 64))
+        self.exposure = afwImage.ExposureF(maskedImage, exposureInfo)
+
+    def test_run(self):
+        # Run with default trail length
+
+        trailedSourceFilterTask = TrailedSourceFilterTask()
+
+        results = trailedSourceFilterTask.run(self.diaSources, self.exposure)
+
+        self.assertEqual(len(results.diaSources),
+                         3)
+
+    def test_run_short_max_trail(self):
+        # Run with low trail rate cutoff to test config settings.
+
+        config = TrailedSourceFilterTask.ConfigClass()
+        config.maxTrailLength = 0.01
+        trailedSourceFilterTask = TrailedSourceFilterTask(config=config)
+
+        results = trailedSourceFilterTask.run(self.diaSources, self.exposure)
+
+        self.assertEqual(len(results.diaSources),
+                         1)
+
+    def test_run_no_trails(self):
+        # Run with long trail length rate so that no sources are filtered.
+
+        config = TrailedSourceFilterTask.ConfigClass()
+        config.maxTrailLength = 10.00
+        trailedSourceFilterTask = TrailedSourceFilterTask(config=config)
+
+        results = trailedSourceFilterTask.run(self.diaSources, self.exposure)
+
+        self.assertEqual(len(results.diaSources),
+                         5)
+        self.assertEqual(len(results.trailedDiaSources),
+                         0)
+
+    def test_check_dia_source_trail(self):
+        # Check that the task outputs the correct masks.
+
+        trailedSourceFilterTask = TrailedSourceFilterTask()
+        mask = trailedSourceFilterTask.check_dia_source_trail(self.diaSources, self.exposure)
+
+        comparison = mask == [False, False, False, True, True]
+        self.assertTrue(comparison.all())
```

Comment:
This makes for a more useful comparison:

```suggestion
        np.testing.assert_array_equal(mask, [False, False, False, True, True])
```

Reply:
Swapped to your suggested code.